encoding/basics-and-workflow.md

```diff
@@ -2,36 +2,584 @@
+that you will use:
 
+* FLAC
+* AAC
+* opus
+
+FLAC is a **F**ree **L**ossless **A**udio **C**odec.
+This is the go-to for lossless audio encoding,
+as it's better at compression
+than other lossless codecs (like PCM).
+You can encode to fLAC using ffmpeg
+with the following command:
+
+`$ ffmpeg -i input.wav output.flac`
+
+You can also change the compression level.
+What this does is that it will try to compress
+the audio file even further than with default settings.
+However, this will take more resources to encode *and* decode.
+Levels range from 0 to 12.
+
+`$ ffmpeg -i input.wav -compression_level 12 output.flac`
+
+To show the difference in filesize,
+I will use [this](audio/Fate_Zero_OP2_(original).wav) audio file
+as an example.
+This was ripped straight from the Fate/Zero blu-rays.
+The first part is the filesize in bytes,
+and the second is the name of the file.
+The number in the brackets
+indicates the `compression_level` used.
+
+```
+26.811.464 Fate_Zero_OP2_(original).wav
+20.373.919 Fate_Zero_OP2_(00).flac
+20.324.721 Fate_Zero_OP2_(01).flac
+20.313.009 Fate_Zero_OP2_(02).flac
+19.672.150 Fate_Zero_OP2_(03).flac
+19.602.556 Fate_Zero_OP2_(04).flac
+19.600.585 Fate_Zero_OP2_(05).flac
+19.599.423 Fate_Zero_OP2_(06).flac
+19.675.564 Fate_Zero_OP2_(07).flac
+19.526.067 Fate_Zero_OP2_(08).flac
+19.585.149 Fate_Zero_OP2_(09).flac
+19.525.085 Fate_Zero_OP2_(10).flac
+19.509.644 Fate_Zero_OP2_(11).flac
+19.469.016 Fate_Zero_OP2_(12).flac
+```
+
+The difference between them for something as short as an Opening
+isn't that significant,
+but that changes
+once you start looking at full episodes:
+
+```
+458.233.964 KaguyaBD_01_(original).wav
+314.660.781 KaguyaBD_01_(00).flac
+312.773.683 KaguyaBD_01_(01).flac
+312.486.815 KaguyaBD_01_(02).flac
+300.712.094 KaguyaBD_01_(03).flac
+299.806.321 KaguyaBD_01_(04).flac
+299.658.920 KaguyaBD_01_(05).flac
+299.606.279 KaguyaBD_01_(06).flac
+300.119.565 KaguyaBD_01_(07).flac
+298.832.315 KaguyaBD_01_(08).flac
+299.172.074 KaguyaBD_01_(09).flac
+298.773.829 KaguyaBD_01_(10).flac
+298.468.836 KaguyaBD_01_(11).flac
+297.952.201 KaguyaBD_01_(12).flac
+```
+
+We can drastically reduce the filesize
+by making use of `-sample_fmt s16`.
+Not a lot of anime have true 24bit audio,
+and the vast majority of people
+wouldn't be able to tell a difference regardless.
+Some encoders prefer encoding FLAC in 16bit instead
+for this very reason.
+
+```
+458.233.964 KaguyaBD_01_(original).wav
+163.088.307 KaguyaBD_01_(00)_16bit.flac
+161.313.003 KaguyaBD_01_(01)_16bit.flac
+161.051.956 KaguyaBD_01_(02)_16bit.flac
+149.929.246 KaguyaBD_01_(03)_16bit.flac
+149.084.733 KaguyaBD_01_(04)_16bit.flac
+148.909.370 KaguyaBD_01_(05)_16bit.flac
+148.857.694 KaguyaBD_01_(06)_16bit.flac
+148.848.250 KaguyaBD_01_(07)_16bit.flac
+148.149.100 KaguyaBD_01_(08)_16bit.flac
+148.088.813 KaguyaBD_01_(09)_16bit.flac
+148.080.093 KaguyaBD_01_(10)_16bit.flac
+147.368.127 KaguyaBD_01_(11)_16bit.flac
+147.327.448 KaguyaBD_01_(12)_16bit.flac
```

Comment:
I don't think we need this many example files for the sake of comparing audio file sizes.